public/static/js/projects.js

```diff
@@ -560,21 +556,15 @@ mciModule.controller('ProjectCtrl', function($scope, $window, $http, $location,
     return true;
   }
 
-  $scope.validPatchDefinition = function(variantRegex, taskRegex){
-    if (!variantRegex || !taskRegex){
-      return false;
-    }
-
-    return true;
-  };
-
-  $scope.validPatchAlias = function(alias, variantRegex, taskRegex, taskTags){
-    if (!alias || !variantRegex || (!taskRegex && !taskTags)){
-      return false;
-    }
+  $scope.validPatchDefinition = function(alias){
+    // variant AND (task XOR tags_
+    return alias && alias.variant && (Boolean(alias.task) != !_.isEmpty(alias.tags))
+  }
 
-    return true;
-  };
+  $scope.validPatchAlias = function(alias){
```

Comment:
Is this necessary? I thought (but have not verified) the github alias also has an implicit alias.alias of "__github"

Reply:
Probably, I already answered this question in a comment above. This confusion happen due to meaningful variable name `*alias`, which may mean string property (like `__github`) or an `Object{alias, variant, task, tags}` in different contexts. This validation is absolutely required and it does nothing with `__github` aslias.alias, as you mentioned above.